### PR TITLE
new model;

### PR DESCRIPTION
**Pull request by olegovna-galina**

### CODE REVIEW

File: src/main/java/com/solutions/shop/model/Customer.java [hunk 1]
```diff
@@ -7,13 +7,22 @@
...
+import javax.persistence.MapsId;
 
 @Data
 @Entity
 @Table(name = "customer")
 public class Customer {
-    @Id
-    @Column(name = "customer_id")
+    //@Id
+    // @Column(name = "customer_id")
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/model/Customer.java]: Remove commented lines

File: src/main/java/com/solutions/shop/model/Product.java [hunk 2]
```diff
@@ -15,7 +13,7 @@
 @Table(name = "product")
 public class Product {
     @Id
-    @Column(name = "product_id")
+    // @Column(name = "product_id")
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/model/Product.java]: Remove commented line

File: src/main/java/com/solutions/shop/model/ShoppingCart.java [hunk 3]
```diff
@@ -7,19 +7,36 @@
...
+import javax.persistence.JoinColumn;
+import java.util.List;
 
 @Data
 @Entity
 @Table(name = "shopping_cart")
 public class ShoppingCart {
     @Id
-    @Column(name = "order_id")
+    // @Column(name = "order_id")
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/model/ShoppingCart.java]: Remove

File: src/main/java/com/solutions/shop/model/ShoppingCart.java [hunk 4]
```diff
@@ -7,19 +7,36 @@
...
     private Integer orderId;
 
+    /*
     @Column(name = "customer_id")
     @NotNull
-    private Integer customerId;
+    */
+    @OneToMany (mappedBy="customer_id", fetch=FetchType.EAGER)
+    // private Integer customerId;
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/model/ShoppingCart.java]: Remove commented

File: src/main/java/com/solutions/shop/model/User.java [hunk 5]
```diff
@@ -13,7 +16,9 @@
 @Table(name = "users")
 public class User {
     @Id
-    @Column(name = "user_id")
+    // @Column(name = "user_id")
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/model/User.java]: Remove commented line

**Review by AlecKotovichSAM — CHANGES_REQUESTED** [review]

File: src/main/java/com/solutions/shop/model/Customer.java [hunk 6]
```diff
@@ -24,15 +25,24 @@ public class Customer {
...
 
     @Column(name = "address")
+    @NotBlank
     private String address;
 
-    public Customer(Integer customerId) {
+    public Customer(){
+    }
+
+    public Customer(Integer customerId, String firstName, String lastName, String address) {
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/model/Customer.java]: What do you think about starting using lombok? You could use here @RequiredArgsConstructor annotation

File: src/main/java/com/solutions/shop/model/Product.java [hunk 7]
```diff
@@ -35,6 +35,15 @@ public class Product {
     @NotBlank
     private String category;
 
+    public Product(Integer productId, String name, String description, BigDecimal price, String brand, String category) {
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/model/Product.java]: lombok? You could use here @RequiredArgsConstructor annotation

File: src/main/java/com/solutions/shop/repository/ICustomerRepository.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.solutions.shop.repository;
+
+import com.solutions.shop.model.Customer;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ICustomerRepository extends JpaRepository<Customer, Integer> {
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/repository/ICustomerRepository.java]: Ask @jsdima maybe you can remove I letter in interface name start

File: src/main/java/com/solutions/shop/repository/IProductRepository.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.solutions.shop.repository;
+
+import com.solutions.shop.model.Product;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface IProductRepository extends JpaRepository<Product, Integer> {
```
Comment (AlecKotovichSAM) [inline src/main/java/com/solutions/shop/repository/IProductRepository.java]: the same

File: src/test/java/com/solutions/shop/repository/CustomerRepositoryTest.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+
+@DataJpaTest
+public class CustomerRepositoryTest {
+    @Autowired
+    private TestEntityManager entityManager;
+
+    @Test
+    public void testCustomerCreate() {
+        User user1 = new User(998,"test-logon", "test-passw");
```
Comment (AlecKotovichSAM) [inline src/test/java/com/solutions/shop/repository/CustomerRepositoryTest.java]: This test tests C in CRUD. What about RUD tests?

File: src/test/java/com/solutions/shop/repository/ProductRepositoryTest.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+import com.solutions.shop.model.Product;
+import java.math.BigDecimal;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+
+@DataJpaTest
+public class ProductRepositoryTest {
```
Comment (AlecKotovichSAM) [inline src/test/java/com/solutions/shop/repository/ProductRepositoryTest.java:13]: extend tests with full CRUD

File: src/test/java/com/solutions/shop/repository/ShoppingCartRepositoryTest.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+import java.math.BigDecimal;
+
+import com.solutions.shop.model.User;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+
+@DataJpaTest
+public class ShoppingCartRepositoryTest {
```
Comment (AlecKotovichSAM) [inline src/test/java/com/solutions/shop/repository/ShoppingCartRepositoryTest.java:18]: the same

**Review by AlecKotovichSAM — CHANGES_REQUESTED** [review]

File: src/main/java/com/solutions/shop/model/Product.java (hunk 7 above)
Comment (olegovna-galina) [reply src/main/java/com/solutions/shop/model/Product.java]: I want to have the Constructor without shoppingCart. Admin should create all products independently.

File: src/main/java/com/solutions/shop/repository/ICustomerRepository.java (hunk 8 above)
Comment (olegovna-galina) [reply src/main/java/com/solutions/shop/repository/ICustomerRepository.java]: @jsdima, should I remove "I" letter in interface name?

File: src/test/java/com/solutions/shop/repository/CustomerRepositoryTest.java (hunk 10 above)
Comment (olegovna-galina) [reply src/test/java/com/solutions/shop/repository/CustomerRepositoryTest.java]: later  :)

File: src/main/java/com/solutions/shop/model/Product.java (hunk 7 above)
Comment (AlecKotovichSAM) [reply src/main/java/com/solutions/shop/model/Product.java]: Lombok @RequiredArgsConstructorc generates a constructor with required arguments. Required arguments are uninitialized final fields and fields with constraints such as @NonNull. So you can make shoppingCart non required

**Review by AlecKotovichSAM — CHANGES_REQUESTED** [review]
check unresolved conversations

File: src/test/java/com/solutions/shop/repository/ProductRepositoryTest.java (hunk 11 above)
Comment (olegovna-galina) [reply src/test/java/com/solutions/shop/repository/ProductRepositoryTest.java:13]: Done. I'm not sure only, if I correctly Unit-test for "delete" made.

**Review by AlecKotovichSAM — APPROVED** [review]

**Pull request merged**